### PR TITLE
Analyzer load failure suppress

### PR DESCRIPTION
**Pull request by chsienki**
Fixes the compiler side of https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1434159 (can't suppress `CS8034`)

We were already correctly suppressing the given warning when using `/noWarn` but weren't respecting options set in a global config.

### CODE REVIEW

**Comment by chsienki** [comment]
@mavasani or @jasonmalinowski could you look into the IDE side of things? I took a quick look and it seems like it probably needs to be handled here https://github.com/dotnet/roslyn/blob/main/src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioAnalyzer.cs#L63-L68 but I'm not familiar with how the IDE wires it all up.

**Comment by chsienki** [comment]
@dotnet/roslyn-compiler for review please

**Comment by jasonmalinowski** [comment]
@chsienki I'm kinda surprised we're wanting people to suppress this at all; can't they fix the underlying problem instead?

**Comment by chsienki** [comment]
@jasonmalinowski This is being caused by an external tool (see https://github.com/fsprojects/Paket/issues/3877) so the user wants to be able to ignore the warning that occurs when trying to load a non-analyzer object.

Agree its kind of a weird edge case, but its also weird that its only a warning but you can't suppress it.

**Comment by jasonmalinowski** [comment]
Hmm, then yeah this might be a bit strange to figure out how to wire up. The strange part here is the code that is dealing with analyzer loading errors isn't really in the context of a Compilation at all, so we'd have to thread through and also deal with the options changing. This area kinda needs a rewrite, and this might be an good forcing function for it.

Either way though how do you want to deal with sequencing? Do you want to merge this and let the IDE do the follow-up work or try doing both pieces at once?

File: src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs [hunk 1]
```diff
@@ -14261,6 +14261,46 @@ public void TestDuplicateAdditionalFiles(string additionalFilePath1, string addi
         [InlineData("abc/a.txt", "./../ABC/a.txt", 2)]
         public void TestDuplicateAdditionalFiles_Linux(string additionalFilePath1, string additionalFilePath2, int expectedCount) => TestDuplicateAdditionalFiles(additionalFilePath1, additionalFilePath2, expectedCount);
 
+        [Fact]
+        public void CanSuppressAnalyzerLoadWarning()
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs:14265]: Consider adding a corresponding VB test

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 2)

**Comment by jcouv** [comment]
@dotnet/roslyn-compiler for second review. Thanks

File: src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs [hunk 2]
```diff
@@ -14261,6 +14261,46 @@ public void TestDuplicateAdditionalFiles(string additionalFilePath1, string addi
         [InlineData("abc/a.txt", "./../ABC/a.txt", 2)]
         public void TestDuplicateAdditionalFiles_Linux(string additionalFilePath1, string additionalFilePath2, int expectedCount) => TestDuplicateAdditionalFiles(additionalFilePath1, additionalFilePath2, expectedCount);
 
+        [Fact]
```
Comment (jjonescz) [inline src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs:14264]:
```suggestion
        [Fact, WorkItem(1434159, "https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1434159")]
```

File: src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs [hunk 3]
```diff
@@ -14261,6 +14261,46 @@ public void TestDuplicateAdditionalFiles(string additionalFilePath1, string addi
...
+
+            // supressed
+            VerifyOutput(dir, src, additionalFlags: new[] { "/analyzer:" + notAnalyzer.Path, "/nowarn:CS8034" }, expectedWarningCount: 0, includeCurrentAssemblyAsAnalyzerReference: false);
+
+            // elevated
+            output = VerifyOutput(dir, src, additionalFlags: new[] { "/analyzer:" + notAnalyzer.Path, "/warnAsError:CS8034" }, expectedErrorCount: 1, includeCurrentAssemblyAsAnalyzerReference: false);
+            Assert.Contains("error CS8034", output, StringComparison.Ordinal);
+        }
+
+        [Fact]
```
Comment (jjonescz) [inline src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs:14285]:
```suggestion
        [Fact, WorkItem(1434159, "https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1434159")]
```

**Review by jjonescz — APPROVED** [review]

**Pull request merged**